runtime/runtime/src/ext.rs

```diff
@@ -389,4 +445,31 @@ impl<'a> External for RuntimeExt<'a> {
             .validator_total_stake(self.epoch_id, self.prev_block_hash)
             .map_err(|e| ExternalError::ValidatorError(e).into())
     }
+
+    #[cfg(feature = "protocol_feature_function_call_ratio")]
+    fn distribute_unused_gas(&mut self, gas: u64) -> u64 {
+        if self.gas_ratio_sum != 0 {
+            let gas_per_ratio = gas / self.gas_ratio_sum;
+
+            self.distribute_leftover_gas_to
+                .drain(..)
+                .map(|GasRatioMetadata { receipt_index, action_index, gas_ratio }| {
+                    let assign_gas = gas_per_ratio * gas_ratio;
+                    if let Some(Action::FunctionCall(FunctionCallAction { ref mut gas, .. })) = self
+                        .action_receipts
+                        .get_mut(receipt_index)
+                        .and_then(|(_, receipt)| receipt.actions.get_mut(action_index))
+                    {
+                        *gas += assign_gas;
+                    } else {
+                        panic!("Invalid index for assigning unused gas ratio");
+                    }
+
+                    assign_gas
```

Comment:
It is preferable if we always distribute all of the unused gas to avoid emitting gas refund receipt. It is okay if the last receipt gas a little bit extra gas. Right now we have leftover gas because of the division. We then would need to write tests to make sure refund receipt is never emitted. Once we do it the return value is always going to be either equal to `gas` or zero, which means it will become semantically a bool.

Reply:
The only issue I see to this is the overhead of getting the modulus of the division before the operations or having to re-index the initial `gas % n` items if we want to micro optimize. Will do this change and then will re-visit later to see if implications on the cost of the operation

Reply:
Made changes with Max's suggestion for now for simplicity, but will update once we resolve these edge cases :)